### ISSUE TITLE
[EPIC] Implement pedestrian view - tool

### ISSUE THREAD

**Issue by AnaBelgun** [TerriaJS/terriajs#5019]
### Problem Statement
_What problem or need are we trying to solve or fulfil?_

**Speculative / to be validated:**

How might we help **urban planners**(?) remotely assess and communicate the spatial impact of buildings/infrastructure on pedestrians?

### Impact
_What's the impact of this problem on our customers and to our business?_

**Speculative / to be validated:**

- Negative spatial properties of buildings/infrastructure (past/present/future) are less likely to be picked up early in the planning process, resulting in lost time and money
- Stakeholders have less visibility over the planning process, resulting a less diverse feedback
- Planners are required to use expensive and/or complex 3D/GIS software to simulate buildings/infrastructure from the perspective of a pedestrian

### How do we judge success?
_What are the goals of the project and the success criteria by which they will be measured? These need to be specific and measurable._

**Refine with client**
1. Number of visitors independently activate and engage with this tool (Google Analytics)
2. Number of planners who report using this tool to assess and/or communicate buildings/infrastructure proposals
3. Positive feedback from users (via interviews, email, map feedback tool etc.)

### What do we already know?
_What data or insights do we have to validate the problem?_

**Speculative examples to prompt data collection:**

**Anecdotes from client or target user group**
- Expensive/time consuming pedestrian POV analysis?
- Difficulty communicating pedestrian POV to others e.g. stakeholders unable to review proposals remotely?
- Negative spatial impacts of new buildings/infrastructure missed in planning stage? Cost of changes?

**Competitor analysis**
- Who else is already supporting this use case or offering similar functionality?
- What are the gaps in their offering?
- … ?

**Visitor analytics**
- number of visitors who have added reality mesh data or 3D model to the map in the past month?
- ... ?

### What do we need to answer?
_What assumptions are we making that need to be validated/refuted?_

- Users can access either Chrome, Firefox or Edge browser…
- Users have access to a keyboard and mouse and are able to use them simultaneously
- … ?

---

User comment: 
"put the two points (location and direction) on the map to create a view from our viewpoint. To expand this capability to create a walking view mode. Basically a user can draw a line and simulate a walking view along the path. 
Inspiration: https://kartta.hel.fi/3d/#/
![image](https://user-images.githubusercontent.com/18452161/100305393-1c1ba980-2ff5-11eb-8315-03ca63e8d0be.png)

**Comment by zoran995** [comment]
Swiss ([geoadmin](https://map.geo.admin.ch/?lang=en&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swissnames3d&layers_visibility=false,false,false,true&layers_timestamp=18641231,,,&lon=7.44946&lat=46.95271&elevation=690&heading=14.578&pitch=-1.803)) has something similar to this in 3D mode ([here is the help](https://help.geo.admin.ch/?id=28&lang=en) for the functions they support in this mode). It is something like AR version for both desktop and mobile

**Comment by AnaBelgun** [comment]
great examples @zoran995 thank you; @philipgrimmett check this out 👍

**Comment by philipgrimmett** [comment]
While we're waiting on use cases from the client to refine this issue, some fairly generic concept sketches to help us generate our own (speculative, will need to be validated): https://invis.io/GWZXKYI4P6V#/442408592_pedestrian_-_1-1

Covered in this prototype:

**Enabling pedestrian mode**
Tool icon appears in close proximity to existing camera controls

**Selecting a drop point / canceling selection prompt**
How do we avoid dropping users inside of 3D object? Land on highest point?

**Locomotion controls overview window**
Controls overview assumes users have access to a mouse/trackpad and keyboard.

**WASD compass(?)**
Early concept similar to Helsinki that allows for basic locomotion (WASD) and compass rotation without the use of a keyboard. Needs more discussion. Replaced default compass/gyro because that functionality is controlled by mouse+drag in pedestrian mode.

**Removed zoom buttons**
Might get confusing to have zoom enabled while in pedestrian mode (locomotion speed etc.), so removed until we have a use case for wide-angle shots. 

**Minimap**
An attempt to resolve the issue of users zooming down to street level and immediately losing their bearings. Also opens up the possibility of quick jumps across large areas. Location icon shows camera direction and zoom angle(?)

**Exiting pedestrian mode**
Needs testing in the PoC, but currently pedestrian mode can be exited (back to position when it was triggered?) by clicking on the pedestrian icon (gif below). We might need to be provide a more visible 'exit' button, but this should be fine for the moment. 

![terria-exitPedestrianMode](https://user-images.githubusercontent.com/5292858/105291371-b884b580-5c0c-11eb-8d93-04a4bc8472bb.gif)

**Comment by na9da** [comment]
Follow up issues:

- [ ] https://github.com/TerriaJS/terriajs/issues/5192
- [x] https://github.com/TerriaJS/terriajs/issues/5193
- [x] https://github.com/TerriaJS/terriajs/issues/5194
- [ ] https://github.com/TerriaJS/terriajs/issues/5196
- [ ] https://github.com/TerriaJS/terriajs/issues/5197
- [ ] https://github.com/TerriaJS/terriajs/issues/5200
- [ ] https://github.com/TerriaJS/terriajs/issues/5201
- [x] https://github.com/TerriaJS/terriajs/issues/5202

**Comment by zoran995** [comment]
@na9da Nice start towards implementing a pedestrian mode. My two cents on this, personally, it feels like lacking a major pedestrian thing which is walking (fixing the user over terrain), currently feels like flying. Some of the features implemented could be useful as a general feature, keyboard events to move map, mini-map (not in its current position).

**Comment by na9da** [comment]
Hi @zoran995, yes we are looking to implement an option for the pedestrian to follow the ground - https://github.com/TerriaJS/terriajs/issues/5227. 

The minimap & movements controller are currently implemented in the context of pedestrian mode but can be extracted out for other use cases.

**Comment by philipgrimmett** [comment]
Minor styling ticket when you get the chance @na9da: https://github.com/TerriaJS/terriajs/issues/5318

**Comment by AnaBelgun** [comment]
v1 implemented. 
to open a new ticket with v2 when needed
